modules/querier/http.go

```diff
@@ -109,6 +106,14 @@ func validateAndSanitizeRequest(r *http.Request) (string, string, bool, error) {
 		return "", "", false, fmt.Errorf("invalid value for queryIngesters %s", q)
 	}
 
+	// no need to validate/sanitize other parameters if queryIngesters is true
+	if queryIngesters {
```

Comment:
the querier should not quit validating if queryIngesters is true.  Yes, the query frontend no longer constructs queries that have both queryIngesters and block ids, but it's still a valid query that the querier has to take.

Reply:
Right, but we wouldn't be using the other parameters right? If they're empty they're set to `blockIDMin` and `blockIDMax`.. and I personally like that behaviour

Reply:
Right now if you curl a querier at `/querier/api/traces/{traceID}` it will do what I expect and check the ingesters and all blocks.  With these changes I think queryIngesters would be set to true and this would only query the ingesters.  I would prefer keeping the previous behavior.